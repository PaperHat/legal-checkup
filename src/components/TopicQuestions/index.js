import { map } from 'ramda'
import React from 'react'

import StyledDesktopQuestionButton from '@components/styled/DesktopQuestionButton'

import StyledTopicName from '@components/styled/TopicName'

import StyledTopicQuestions from '@components/styled/TopicQuestions'

import makeQuestionButton from '@wrappers/makeQuestionButton'

export default function TopicQuestions ({ topic, topicQuestions, activeTopic }) {
  console.log('compoent', activeTopic)
  return (
    <StyledTopicQuestions activeTopic={activeTopic}>
      <StyledTopicName>{topic}</StyledTopicName>
      <div>
        {map(({ id, index }) => {
          const QuestionButton = makeQuestionButton(StyledDesktopQuestionButton)

          return (
            <QuestionButton key={id} questionIndex={index}>
              {index + 1}
            </QuestionButton>
          )
        }, topicQuestions)}
      </div>
    </StyledTopicQuestions>
  )
}
