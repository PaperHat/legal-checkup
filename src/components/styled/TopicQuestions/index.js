import styled from 'styled-components'

const TopicQuestions = styled.div`
  font-weight: ${({ activeTopic }) => (activeTopic ? 'bold' : 'normal')};

  background-color: #fff;
  display: inline-block;
`

TopicQuestions.displayName = 'StyledTopicQuestions'

export default TopicQuestions
