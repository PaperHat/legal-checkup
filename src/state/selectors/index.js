export function getTestCount ({ rootReducer: { count } = {} }) {
  return count
}
