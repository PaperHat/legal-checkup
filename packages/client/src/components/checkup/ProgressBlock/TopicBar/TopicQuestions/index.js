import styled from 'styled-components'

const TopicQuestions = styled.div`
  background-color: #ededed;
  font-weight: ${({ activeTopic }) => (activeTopic ? 'bold' : 'normal')};
  margin-top: -55px;
  margin-bottom: 20px;
  margin-left: -5px;
  margin-right: 5px;
  background-color: transparent;
  display: inline-block;
  color: ${({ activeTopic }) => (activeTopic ? '#b62645' : '#ededed')};
`

TopicQuestions.displayName = 'StyledTopicQuestions'

export default TopicQuestions
