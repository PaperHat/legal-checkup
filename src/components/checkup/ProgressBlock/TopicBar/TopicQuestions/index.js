import styled from 'styled-components'

const TopicQuestions = styled.div`
  background-color: #ededed;
  font-weight: ${({ activeTopic }) => (activeTopic ? 'bold' : 'normal')};
  margin-top: -55px;
  margin-bottom: 20px;
  margin-left: -5px;
  margin-right: 5px;
  display: inline-block;
  color: ${({ activeTopic }) => (activeTopic ? '#b62645' : '#a1a1a1')};
`

TopicQuestions.displayName = 'StyledTopicQuestions'

export default TopicQuestions
