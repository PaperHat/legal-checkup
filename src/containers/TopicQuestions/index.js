import { connect } from 'react-redux'

import { makeGetTopicQuestions } from '@state/selectors'
import TopicQuestions from '@components/TopicQuestions'

function makeMapStateToProps () {
  const getTopicQuestions = makeGetTopicQuestions()
  return function mapStateToProps (state, { topic }) {
    return {
      topicQuestions: getTopicQuestions(state, { topic })
    }
  }
}

export default connect(makeMapStateToProps)(TopicQuestions)
