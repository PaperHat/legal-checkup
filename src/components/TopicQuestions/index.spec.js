import React from 'react'
import { shallow } from 'enzyme'

import { state } from '@state/fixtures'
import { getCurrentTopicName, makeGetTopicQuestions } from '@state/selectors'

import TopicQuestions from '.'

const getTopicQuestions = makeGetTopicQuestions()

describe('components:TopicQuestions', () => {
  it('should render the TopicQestions and CSS properties properly', () => {
    expect(toJson(shallow(<TopicQuestions />))).toMatchSnapshot()
  })

  it('should render the TopicQestions and CSS properties properly when activeTopic', () => {
    const topic = getCurrentTopicName(state)

    expect(
      toJson(shallow(<TopicQuestions activeTopic={true} topic={topic} />))
    ).toMatchSnapshot()
  })

  it('should render the TopicQestions and CSS properties properly when activeTopic', () => {
    const topic = getCurrentTopicName(state)
    const topicQuestions = getTopicQuestions(state, { topic })

    expect(
      toJson(
        shallow(
          <TopicQuestions
            activeTopic={true}
            topic={topic}
            topicQuestions={topicQuestions}
          />
        )
      )
    ).toMatchSnapshot()
  })
})
