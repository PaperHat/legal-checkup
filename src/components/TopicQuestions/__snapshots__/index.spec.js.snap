// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:TopicQuestions should render the TopicQestions and CSS properties properly 1`] = `
<StyledTopicQuestions>
  <StyledTopicName />
  <div />
</StyledTopicQuestions>
`;

exports[`components:TopicQuestions should render the TopicQestions and CSS properties properly when activeTopic 1`] = `
<StyledTopicQuestions
  activeTopic={true}
>
  <StyledTopicName />
  <div />
</StyledTopicQuestions>
`;

exports[`components:TopicQuestions should render the TopicQestions and CSS properties properly when activeTopic 2`] = `
<StyledTopicQuestions
  activeTopic={true}
>
  <StyledTopicName>
    Money Troubles
  </StyledTopicName>
  <div>
    <Connect(StyledDesktopQuestionButton)
      key="759CDC30C25B489A9EB71B7E859F2DD7"
      questionIndex={0}
    >
      1
    </Connect(StyledDesktopQuestionButton)>
    <Connect(StyledDesktopQuestionButton)
      key="8A9B3C23BA41440187B0B9F9EB0D8400"
      questionIndex={1}
    >
      2
    </Connect(StyledDesktopQuestionButton)>
    <Connect(StyledDesktopQuestionButton)
      key="A7D51C38F34A4EA1919C74013E703C39"
      questionIndex={2}
    >
      3
    </Connect(StyledDesktopQuestionButton)>
    <Connect(StyledDesktopQuestionButton)
      key="5DBE75B1D33D43DD8AC4E00730B5BD3D"
      questionIndex={3}
    >
      4
    </Connect(StyledDesktopQuestionButton)>
    <Connect(StyledDesktopQuestionButton)
      key="E2C5A2E126474F56B76719EBC818686E"
      questionIndex={4}
    >
      5
    </Connect(StyledDesktopQuestionButton)>
  </div>
</StyledTopicQuestions>
`;
